R/DevToolsConnexion.R

```diff
@@ -22,40 +26,112 @@ CDPSession <- R6::R6Class(
         # if error, emit an error
         if (!is.null(data$error)) {
           "!DEBUG error: `event$data`"
-          self$emit('error', data$error)
+          self$emit("error", paste0("Error: '", data$error$message, "'(code ", data$error$code, ")."))
         }
-        # if a reponse to a command, emit the callback corresponding to the id
+        # if a reponse to a command, emit a response event
         if (!is.null(id)) {
-          method_sent <- private$.commandList[[self$id]]$method
-          private$.commandList[[self$id]] <- NULL
-          self$emit(method_sent, data)
+          self$emit("response", id = data$id, result = data$result)
         }
         # if an event is fired, emit the corresponding listeners
         if (!is.null(method)) {
-          self$emit(method, data)
+          self$emit(method, params = data$params)
         }
       })
       ws$onClose(function(event) {
         "!DEBUG R disconnected from headless Chrome with code `event$code`"
         "!DEBUG and reason `event$reason`."
         # later::later(~ chr_close(chr_process, work_dir), delay = 0.2)
+        self$emit("disconnect")
```

Comment:
No need to have `self$emit("disconnect", self)` here like for `connect` event ?

Reply:
I just adopted the same flavor than here: https://github.com/cyrus-and/chrome-remote-interface#event-disconnect